build/src/VersionParser.php

```diff
@@ -29,10 +31,12 @@ public static function handleCore($version) {
     return ">=$major.$minor,<$version";
   }
 
-  public static function handleContrib($version) {
-    list($core, $version) = explode('-', $version, 2);
-    list($major) = explode('.', $version);
-    return ">=$major,<$version";
+  public static function handleContrib($version, $result) {
+    // $result->taxonomy_vocabulary_6 is usually a term like 8.x (https://www.drupal.org/taxonomy/term/7234).
+    // Its absence indicates a semver release (or a core release).
+    list($core, $version) = empty($result->taxonomy_vocabulary_6) ? [NULL, $version] : explode('-', $version, 2);
+    list($major, $minor) = explode('.', $version, 2);
+    return ">=$major.$minor,<$version";
```

Comment:
I believe API-compatibility-prefixed version numbers still need the original handling, ">=$major,<$version"

This really isn't core vs. contrib, it is major.minor vs. major.minor.patch. If this ever handled D7 core, that would also want ">=$major,<$version", since that was also not semver.

Reply:
>  API-compatibility-prefixed version numbers

And whats the best way to identify those?

Reply:
@drumm and we do have to handle D7 core in this package. Drush doesn't use it but others might.

Reply:
Once `list($core, $version) = empty($result->taxonomy_vocabulary_6) ?  …` is done for contrib, using the original version number for core, since there is already no API compatibility prefix - something like:

```
$version_components = explode('.', $version);
if (count($version_components) === 2) {
  // Only major.minor, either Drupal core 7, or contrib that had an API compatibility prefix.
  list($major) = $version_components;
  return ">=$major,<$version";
}
elseif (count($version_components) === 3) {
  // Semver, either Drupal core 8 or later, or contrib using semver.
  list($major, $minor) = $version_components;
  return ">=$major.$minor,<$version";
}
else {
  // Should not happen.
}
```

Reply:
Thanks. Thats now in the PR. I reran the build and now you can compare the PR output versus the last one run by CircleCI - https://github.com/drupal-composer/drupal-security-advisories/compare/3b3a391...8.x-v2

- Does semver_example row look right to you?
- I dunno why some packages are being removed.

Reply:
semver_example & drupal look okay to me.

I can't see any reason for the other packages to be removed. There are definitely still the same security releases for google_analytics: https://www.drupal.org/project/google_analytics/releases?version=8.x-2.

Reply:
I cleared local cache and reran the build. The packages like google_analytics are no longer removed. Dunno what happenned earlier but it was prob a local problem.